components/carousel/script.js

```diff
@@ -0,0 +1,368 @@
+     * @param {number} direction - 1 for right arrow, -1 for left.
+     * @returns {undefined}
+     */
+    hideArrow(direction) {
+        if (direction === 1) return this.rightArrow.classList.add('guic-carousel-hidden');
+        this.leftArrow.classList.add('guic-carousel-hidden');
+    }
+
+    /**
+     * Show one of the navigation arrows.
+     * @param {number} direction - 1 for right arrow, -1 for left.
+     * @returns {undefined}
+     */
+    showArrow(direction) {
+        if (direction === 1) return this.rightArrow.classList.remove('guic-carousel-hidden');
+        this.leftArrow.classList.remove('guic-carousel-hidden');
+    }
+
+    /**
+     * Check if the next item will overflow the carousel's boundaries.
+     * @param {number} itemsLength - the maximum number of items in the carousel.
+     * @param {number} next - the index of the next element
+     * @returns {boolean}
+     */
+    overflows(itemsLength, next) {
+        return itemsLength - next < this.pageSize;
+    }
+
+    /**
+     * Get the index of the next element based on the current step size.
+     * @param {number} stepSize
+     * @returns {number}
+     */
+    nextStepIndex(stepSize) {
+        return (this.current + stepSize) % this.items.length;
+    }
+
+    /**
+     * Get the index of the previous element based on the current step size.
+     * @param {number} stepSize
+     * @returns {number}
+     */
+    prevStepIndex(stepSize) {
+        const itemsLength = this.items.length;
+        return (this.current - stepSize + itemsLength) % itemsLength;
+    }
+
+    /**
+     * Check if the up nav arrow should be visible - if navigating up is possible.
+     * If not - hide it.
+     * @returns {undefined}
+     */
+    shouldShowUpNavArrow() {
+        if (this.overflows(this.items.length, this.nextStepIndex(this.navArrowStepSize))) return this.hideArrow(1);
+        this.showArrow(1);
+    }
+
+    /**
+     * Check if the down nav arrow should be visible - if navigating down is possible.
+     * If not - hide it.
+     * @returns {undefined}
+     */
+    shouldShowDownNavArrow() {
+        if (this.overflows(this.items.length, this.prevStepIndex(this.navArrowStepSize))) return this.hideArrow(-1);
+        this.showArrow(-1);
+    }
+
+    /**
+     * Add an item to the carousel.
+     * @param {HTMLElement} node
+     * @param {index} index - the position at which to add it.
+     */
+    addItem(node, index) {
+        const contentWrapper = this.querySelector(CONTENT_WRAPPER_SELECTOR);
+        if (index === undefined || index < 0) {
+            contentWrapper.append(node);
+        } else {
+            const referenceNode = this.items[index - 1];
+            referenceNode.parentNode.insertBefore(node, referenceNode);
+        }
+
+        this.createPaginationControls();
+    }
+
+    /**
+     * Remove an item from the carousel.
+     * @param {number} index - the position from which to remove.
+     */
+    removeItem(index = 0) {

```

Comment:
For next tier/s - item can be removed with `node` also.

Reply:
Don't understand what you mean. Can you clarify so that I can add it to the document?

Reply:
The idea is to remove item from the carousel by passing a ref to an HTMLElement. For example, I can do
```
const elementFromCarousel = document.querySelector('.item-1');
carousel.removeItem(elementFromCarousel);
```

The thing here is that I should not know which is the index of my element in the carousel so I can remove it. Instead, the carousel will do it automatically for me.

Reply:
Yes, a lot better - adding in to next tier.